back/engines/commercial/report_builder/app/controllers/report_builder/web_api/v1/reports_controller.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+module ReportBuilder
+  module WebApi
+    module V1
+      class ReportsController < ::ApplicationController
```

Comment:
Just to be clear, this endpoint wraps the entire content builder layouts end point with a superset of functionality so within report builder UI we switch to using only the reports endpoint to save everything.

Reply:
Not sure of what you mean by "wraps the entire content builder layouts end point with a superset of functionality" but yes, the endpoints implemented by this controller should be sufficient to implement the report builder UI.